<launch>
  <arg name="gui"        default="true"/>
  <arg name="world_file" default="$(env TURTLEBOT_GAZEBO_WORLD_FILE)"/>

  <arg name="base"       value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
  <arg name="battery"    value="$(optenv TURTLEBOT_BATTERY /proc/acpi/battery/BAT0)"/>  <!-- /proc/acpi/battery/BAT0 -->
  <arg name="stacks"     value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons -->
  <arg name="3d_sensor"  value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro -->
  <arg name="initial_pose_x" default="2.5"/>
  <arg name="initial_pose_y" default="2.5"/>
  <arg name="initial_pose_a" default="0.0"/>

  <include file="$(find turtlebot_gazebo)/launch/map6_world.launch">
  </include>



  <node name="rviz" pkg="rviz" type="rviz" respawn="true" />

  <!--include file="$(find turtlebot_teleop)/launch/keyboard_teleop.launch"/-->
  <!--include file="$(find turtlebot_gazebo)/launch/gmapping_demo.launch"/-->

  <!-- ************** Navigation *************** -->
  <node pkg="move_base" type="move_base" clear_params="true" respawn="false" name="move_base" output="screen" >

  <rosparam file="$(find trabajo_planificacion)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap"/>

  <rosparam file="$(find trabajo_planificacion)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap"/>

  <rosparam file="$(find trabajo_planificacion)/cfg/local_costmap_params.yaml" command="load" />

  <rosparam file="$(find trabajo_planificacion)/cfg/global_costmap_params.yaml" command="load" />

  <rosparam file="$(find trabajo_planificacion)/cfg/teb_local_planner_params.yaml" command ="load"/>

  <param name="base_global_planner" value="global_planner/GlobalPlanner"/>
  <param name="planner_frequency" value="1.0" />
  <param name="planner_patience" value="5.0" />

  <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/>
  <param name="controller_frequency" value="5.0" />
  <param name="controller_patience" value="15.0" />
  <remap from="/cmd_vel" to="/mobile_base/commands/velocity"/>
  </node>

  <!-- ****** Maps ***** -->
  <node name="map_server" pkg="map_server" type="map_server" args
  ="$(find trabajo_planificacion)/maps/map6.yaml" output="screen">
  <param name="frame_id" value="map"/>
  </node>

<!-- ****** AMCL ***** -->
  <node pkg="amcl" type="amcl" name="amcl" output="screen">
    <param name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <param name="initial_pose_a" value="$(arg initial_pose_a)"/>
    <param name="odom_model_type" value="diff"/>
    <param name="use_map_topic" value="true"/>
    <param name="transform_tolerance" value="0.5"/>
  </node>

</launch>
